### PR TITLE
Dev 1st. Add the base file

### LINKED ISSUE #1: Tag in issue/PR in yui0303
Please see https://github.com/yxuan0329/R-tree-Search-Engine/issues/1 .  It's great to see your code in the repository.  Making PR and tagging me provides me an opportunity to review your code.

### PR DESCRIPTION
**Pull request by yui0303**
The Class Matrix and the related operations in the branch is still developing.
The PR can show the baseline of the term project.

### CODE REVIEW

File: Makefile
Comment (yungyuc) [inline Makefile:1]: Do you know how to set up Github Actions?

File: docker/Dockerfile
Comment (yungyuc) [inline docker/Dockerfile:1]: It is usually hard to use a debugger in a container.  Be careful about it.

File: cpp/_matrix.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include <vector>
+#include <cmath>
+// #include <mkl.h>
+#include <utility>
+#include <iomanip>
+
+// #include <pybind11/pybind11.h>
+// #include <pybind11/stl.h>
+
+class Matrix {
```
Comment (yungyuc) [inline cpp/_matrix.hpp:14]: It will be educating to design a better matrix class.

**Review by yungyuc — APPROVED** [review]
Good PR in early stage.

File: Makefile
Comment (yui0303) [reply Makefile:1]: I'm not familiar with it. Perhaps I need to spend some time researching.

File: docker/Dockerfile
Comment (yui0303) [reply docker/Dockerfile:1]: Oh, I don't realize the problem. I simply want users to set up the environment using Docker.

Comment (yungyuc) [reply docker/Dockerfile:1]: Before you can easily debug and make a decent presentation, it is not realistic to expect to have real users.

File: Makefile
Comment (yungyuc) [reply Makefile:1]: Google it.  It is a common and easy way to implement continuous integration, which boosts your productivity.

File: cpp/_matrix.hpp (hunk 1 above)
Comment (yui0303) [reply cpp/_matrix.hpp:14]: It seems that I can write my Matrix class more effectively. Isn't it?

Comment (yungyuc) [reply cpp/_matrix.hpp:14]: In future PRs, yes.

Comment (yui0303) [reply cpp/_matrix.hpp:14]: Thanks for the reply. I'll keep it in my mind. 
Perhaps I can discover a better approach and present it in the report.

**Comment by yungyuc** [comment]
@yui0303 why don't you merge the PR?

**Pull request merged**